github-actions/feature-request/src/action.ts

```diff
@@ -0,0 +1,210 @@
+ * @param githubIssue Object used for communication with the GitHub API for a particular issue
+ * @param config Configuration of the GitHub bot
+ */
+const processIssue = async (githubAPI: GitHubAPI, githubIssue: GitHubIssueAPI, config: Config) => {
+  const issue = await githubIssue.get();
+
+  // Issues opened by team members bypass the process.
+  if (await githubAPI.isOrgMember(issue.author.name, config.organization)) {
+    log('The creator of this issue is a member of the organization.');
+    return;
+  }
+
+  // An extra assurance we will not get into a situation where we
+  // have issues under consideration / backlog which require votes.
+  if (
+    issue.labels.includes(config.inBacklogLabel) ||
+    issue.labels.includes(config.underConsiderationLabel) ||
+    !issue.labels.includes(config.featureRequestLabel) ||
+    !issue.open
+  ) {
+    log(`Invalid query result for issue #${issue.number}.`);
+    return;
+  }
+
+  if (await shouldConsiderIssue(issue, githubIssue, config)) {
+    log(`Adding #${issue.number} for consideration.`);
+    return githubIssue.addLabel(config.underConsiderationLabel);
+  }
+
+  if (!issue.labels.includes(config.requiresVotesLabel)) {
+    log(`Adding votes required to #${issue.number}`);
+    await githubIssue.addLabel(config.requiresVotesLabel);
+  }
+
+  const timestamps = await getTimestamps(githubIssue);
+
+  if (timestamps.start === null && timestamps.warn === null) {
+    // In case an issue has been open for longer than a specified period of time and
+    // it still does not have enough votes to be under consideration we want to add a warning.
+    if (daysSince(issue.createdAt) >= config.oldIssueWarnDaysDuration) {
+      log(`Adding a warning for old feature request with #${issue.number}`);
+      return await githubIssue.postComment(comment(CommentMarkers.Warn, config.warnComment));
+    }
+
+    // If this is not an old issue, we want to announce the voting process has started.
+    log(`Starting voting for #${issue.number}`);
+    return await githubIssue.postComment(
+      comment(CommentMarkers.StartVoting, config.startVotingComment),
+    );
+  }
+
+  if (
+    timestamps.start !== null &&
+    daysSince(timestamps.start) >= config.warnDaysDuration &&
+    timestamps.warn === null
+  ) {
+    log(`Posting a warning for #${issue.number}`);
+    return await githubIssue.postComment(comment(CommentMarkers.Warn, config.warnComment));
+  }
+
+  if (timestamps.warn !== null && daysSince(timestamps.warn) >= config.closeAfterWarnDaysDuration) {
+    // In the future consider closing associated PRs if we have high
+    // level of confidence they are scoped to the feature request.
+    log(`Closing feature request #${issue.number}`);
+    return await Promise.all([
+      githubIssue.postComment(comment(CommentMarkers.Close, config.closeComment)),
+      config.closeWhenNoSufficientVotes
+        ? githubIssue.close()
+        : githubIssue.addLabel(config.insufficientVotesLabel),
```

Comment:
Just as idea: may be we can add labels in all cases, even if we are about to close the issue? That might help us with introspection later: we should be able to filter all issues/PRs closed with that particular label and:
- do post-process verification of selected issues
- provide various stats, such as avg number of issues close per week, etc